test/unit/test-main.js

```diff
@@ -0,0 +1,136 @@
+var sinon = require('sinon');
+var proxyquire = require('proxyquire');
+var request = require('supertest');
+
+// mock out express app to allow verification in tests
+var mockExpress = {
+  use: function() {},
+  listen: function() {},
+  static: function() {}
+};
+var mockExpressApp = function() {
+  return mockExpress;
+};
+var mockStaticHandler = function static(){};
+mockExpressApp.static = function() {
+  return mockStaticHandler;
+};
+
+// mock out fh-mbaas-api to allow verification in tests
+var mockMbaasApi;
+var mockSys;
+var mockFhMiddleware;
+var mockCloud;
+var mockError;
+var mockSysHandler = function sys(){};
+var mockMbaasHandler = function mbaas(){};
+var mockFhMiddlewareHandler = function fhmiddleware(){};
```

Comment:
This is  an express middleware function so the signature needs to be function(req, res, next){} and it has to call next as a function.